### ISSUE TITLE
Make some docs on how to make a compatibility-patch

### ISSUE THREAD

**Issue by kaptain-kavern** [kaptain-kavern/ADogSaid#38]
#  [A Dog Said (blank) Compatibility Patch Template](https://github.com/kaptain-kavern/ADogSaid/files/1049843/ADogSaid-Compatibility.Patch.Blank.Template.zip)
In this archive you will find a complete pre made mod that will enable **A Dog Said**'s surgeries on your modded animals, all is left to you is to open and correct 2 xml files.         
- in `\Patches\patches.xml`
  - you will need to add your animals `defName`
  - you may have to remove some `<Operation>` node
- Obviously correct values in the `About.xml` as well

This is intended to become a stand-alone mod (**reading** : _you do not included those files in your animal mod_) that you will release alongside your animal mod.

#### [ADogSaid-Compatibility Patch (Blank Template).zip](https://github.com/kaptain-kavern/ADogSaid/files/1049843/ADogSaid-Compatibility.Patch.Blank.Template.zip)

#  More Informations, please read

- You can (and should) rename the mod folder to your needs

- In the patch file, you need to paste your animals `ThingDef`'s `defName` and not the `PawnKindDef`'s ones.

- You definitively can add several animals at once, you just need to put each `defName` between `<li></li>` tags.

- If you don't need some line in the patch (if your animals have no tails or arms), you have to completely remove the whole line. From `<Operation Class="PatchOperationAdd">` to the next `</Operation>`. _You can't comment them, you have to remove them_.

- It seems that patches files aren't taking any comments ( `<!-- -->` ) in any other place than where I already put some (feel free to change them - _Commented code IS good code_ :D )
____________________
Under you'll find a copy/paste from the file provided in the .zip archive
____________________
```xml
<?xml version="1.0" encoding="utf-8" ?>
<Patch>

	<Operation Class="PatchOperationAdd"> <!-- "Inserting animal defNames into surgery recipes == For Organs and all animal parts" -->
		<xpath>/RecipeDefs/RecipeDef[defName = "InstallBionicAnimalStomach" or "InstallBionicAnimalKidney" or "InstallBionicAnimalLiver" or "InstallBionicAnimalLung" or "InstallBionicAnimalHeart" or "InstallAnimalBrainStimulator" or "InstallSpineAnimalBionic" or "InstallSpineAnimalSimple" or "InstallJawAnimalSimple" or "InstallJawAnimalBionic" or "InstallBionicEyeAnimal" or "InstallEarAnimalSimple" or "InstallEarAnimalBionic" or "InstallPegLegAnimal" or "InstallSimpleProstheticLegAnimal" or "InstallBionicLegAnimal"]/recipeUsers</xpath>
		<value>
			<li>YOUR_FIRST_ANIMAL_DEFS_HERE</li>
			<li>YOUR_SECOND_ANIMAL_DEFS_HERE</li>
		</value>
	</Operation>

	<Operation Class="PatchOperationAdd"> <!-- "Inserting animal defNames into surgery recipes == For Tails" -->
		<xpath>/RecipeDefs/RecipeDef[defName = "InstallSimpleProstheticTailAnimal"]/recipeUsers</xpath>
		<value>
			<li>YOUR_FIRST_ANIMAL_DEFS_HERE</li>
			<li>YOUR_SECOND_ANIMAL_DEFS_HERE</li>
		</value>
	</Operation>

	<Operation Class="PatchOperationAdd"> <!-- "Inserting animal defNames into surgery recipes == For Arms" -->
		<xpath>/RecipeDefs/RecipeDef[defName = "InstallSimpleProstheticArmAnimal" or "InstallBionicArmAnimal"]/recipeUsers</xpath>
		<value>
			<li>YOUR_FIRST_ANIMAL_DEFS_HERE</li>
			<li>YOUR_SECOND_ANIMAL_DEFS_HERE</li>
		</value>
	</Operation>

</Patch>
```

**You obviously *may* have to adapt for your animals. Some have tails, some not. Same have arms etc**

**Comment by inkyspoon** [comment]
does this detect automatically if ADS is loaded? wasnt there something that could do that?

or is this...add this if you know ads is loaded?

**Comment by kaptain-kavern** [comment]
like a normal patch ... no ? 👅

**Comment by kaptain-kavern** [comment]
i put the link to zip in front of the doc. 

It's a pre made patch mod. All is left to do is copy in the modded animal defName

**Comment by kd8lvt** [comment]
Err, this is probably a stupid error on my part, but I copied the patches.xml and directories from the zip into my mod file, and only made slight changes (changed 'YOUR_ANIMAL_DEFS_HERE' to my animal defname, and changed the About.xml to better fit the mod) but it's erroring on startup. Attatched is an image of my mod order, as well as my error log.
![screenshot from 2017-06-03 11-34-02](https://cloud.githubusercontent.com/assets/5531395/26754871/c3885e10-4850-11e7-9c62-b5b472081bf0.png)
![screenshot from 2017-06-03 11-34-33](https://cloud.githubusercontent.com/assets/5531395/26754872/c38dd516-4850-11e7-8360-ab7df7d8e873.png)
And here is my file structure (for both mods)
[FerRIMets&ADS-Patch.zip](https://github.com/kaptain-kavern/ADogSaid/files/1049679/FerRIMets.ADS-Patch.zip)

**Comment by kaptain-kavern** [comment]
in your animal code it's ; 
`<defName>ferret</defName>`

so in the patch you should use 

```xml
<value>
	<li>ferret</li>
</value>
```

sorry i'm in a hurry right now. tell me if it works ;)

**Comment by kd8lvt** [comment]
Yeah, I am using KdFerret. I'm using two different def names, ferret and KdFerret (KdFerret is the pawn, ferret is the other thing that I don't know what it's used for, but yorkies had it, so now ferrets do)

So, I have no idea which one I was supposed to use. I'll try just ferret now. (Will be edited to say whether it worked or not)

EDIT:
It worked. Turns out you don't use the kindbase-child def, but the thingbase-child def.
Good to know :P

**Comment by kaptain-kavern** [comment]
So @kd8lvt , i'm back.     
Yeah you figured out, you need to paste your animal `ThingDef/defName` and not the `PawnKindDef`. AFAIK all recipes works like that in vanilla - *In my animal mods I stick to use the same value for animals `ThingDef` and `PawnKindDef` without any problems*

______

Thank you very much for your feedback, I will improve the documentation and code's comments. 
Your help was very much appreciated 👌 

_______
Great job on the mod, codewise i mean, with the namemakers and all 😉  

Cheers 🍷

**Comment by kaptain-kavern** [comment]
@spoonshortage 

# Report : 
- Completely improved the doc (please correct my poor phrasing where it needs 😛 )
- Updated the main post here
- Updated new archive reuploaded (links updated in this page)

_____________

When this seems to be "ideal" I will transfer this doc in it's proper wiki page

**Comment by inkyspoon** [comment]
looks fine 👍
